fuse/namespace.cpp

```diff
@@ -241,38 +235,58 @@ bool create_file(const std::string& dest) {
 // the layered mounts.
 bool do_mounts_from_json(const JAST& jast, const std::string& fuse_mount_path)
 {
+	// Create a known mount-able dir in the parent mount namespace to base the rest of the mounts on.
+	const std::string mount_prefix = "/tmp/.wakebox-mount";
+	if (0 != mkdir(mount_prefix.c_str(), 0777)) {
+		// Re-use existing one as we have our own mount table
+		if (errno != EEXIST) {
+			std::cerr << "mkdir (" << mount_prefix << "): "
+				<< strerror(errno) << std::endl;
+			return false;
+		}
+	}
+
+	bool replacedRoot = false;
+
 	for (auto &x : jast.get("mount-ops").children) {
 		const std::string& op = x.second.get("type").value;
 		const std::string& src = x.second.get("source").value;
 		const std::string& dest = x.second.get("destination").value;
-		const std::string& after_pivot = x.second.get("after-pivot").value;
 		bool readonly = x.second.get("read-only").kind == JSON_TRUE;
 
-		if (!validate_mount(op, src, after_pivot))
-			return false;
+		if (dest == "/")
+			// All mount ops from here onward will have a prefixed destination.
+			// The prefix will be pivoted to after the final mount op.
+			replacedRoot = true;
+
+		const std::string& target = replacedRoot ? (mount_prefix + dest) : dest;
```

Comment:
What happens if the user supplies a path not starting with a "/", like "XYZ"? Looks like this would break and use /tmp/.wakebox-mountXYZ

Reply:
What do we want the behaviour to be in this case? Maybe we should require absolute paths for destination?

Reply:
For destinations before `/` is used, relative works, afterwards they must be absolute within the new mount namespace